<script>
    try {
        // 数値に対して文字列のメソッドであるtoUpperCaseを呼んだ場合
        const num = 1;
        //num.toUpperCase();

        // 存在しない変数val2を参照した場合
        let val1 = val2 + 1;
    } catch(e) {
        if(e instanceof TypeError){
            console.error("TypeErrorが発生した時の処理")
        }else if(e instanceof ReferenceError){
            console.error("ReferenceErrorが発生した時の処理")
        }
    }
</script>

